common/djangoapps/track/tests/test_middleware.py

```diff
@@ -145,3 +145,13 @@ def test_request_headers(self):
             'ip': ip_address,
             'agent': user_agent,
         })
+
+        factory = RequestFactory(REMOTE_ADDR=ip_address, HTTP_USER_AGENT=user_agent, HTTP_X_EDX_GA_CLIENT_ID='123.123')
+        request = factory.get('/some-path')
+        context = self.get_context_for_request(request)
+
+        self.assert_dict_subset(context, {
+            'ip': ip_address,
+            'agent': user_agent,
+            'client_id': '123.123'
+        })
```

Comment:
@aamir-khan fold this into the logic above; it's not necessary to duplicate code like this.